src/java/org/apache/cassandra/db/rows/Cell.java

```diff
@@ -42,8 +44,15 @@
 public abstract class Cell<V> extends ColumnData
 {
     public static final int NO_TTL = 0;
-    public static final int NO_DELETION_TIME = Integer.MAX_VALUE;
-    public static final int MAX_DELETION_TIME = Integer.MAX_VALUE - 1;
+    public static final long NO_DELETION_TIME = Long.MAX_VALUE;
+    public static final int NO_DELETION_TIME_UNSIGNED_INTEGER = CassandraUInt.MAX_VALUE_UINT;
+    public static final long MAX_DELETION_TIME = CassandraUInt.MAX_VALUE_LONG - 2;
+    public static final int MAX_DELETION_TIME_UNSIGNED_INTEGER = CassandraUInt.fromLong(MAX_DELETION_TIME);
```

Comment:
It seems that the only user of `MAX_DELETION_TIME_UNSIGNED_INTEGER` is `DeletionTime`. So we might want to move the constant to that class and make it `private`.

Reply:
I would really prefer to leave it here. Its it's natural place sitting next to it's equivalent and any future users of the constant will look for it here in the first place.

Reply:
Fine with me. I'm starting to think that not only `MAX_DELETION_TIME_UNSIGNED_INTEGER` but all the `*DELETION_TIME*` constants could be more naturally placed in `DeletionTime` than in `Cell`. But that's debatable, and in any case we don't have to do it here.